basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/mongodb/MongoDBAASAPI.java

```diff
@@ -24,92 +24,84 @@
-	protected String collection;
-	protected String aasId;
+	protected String collectionName;
+	private MongoDBBaSyxStorageAPI<AssetAdministrationShell> storageApi;
+	private String shellIdentificationId;
 
 	/**
 	 * Receives the path of the configuration.properties file in its constructor.
 	 * 
 	 * @param config
-	 * @deprecated Use the new constructor using a MongoClient
 	 */
-	@Deprecated
-	public MongoDBAASAPI(BaSyxMongoDBConfiguration config, String aasId) {
-		this(config, aasId, MongoClients.create(config.getConnectionUrl()));
+	public MongoDBAASAPI(BaSyxMongoDBConfiguration config, String shellIdentificationId, MongoClient client) {
+		this(MongoDBBaSyxStorageAPIFactory.<AssetAdministrationShell>create(config.getAASCollection(), AssetAdministrationShell.class, config, client), shellIdentificationId);
+	}
+
+	public MongoDBAASAPI(MongoDBBaSyxStorageAPI<AssetAdministrationShell> mongoDBStorageAPI, String shellIdentificationId) {
+		super();
+		this.storageApi = mongoDBStorageAPI;
+		this.shellIdentificationId = shellIdentificationId;
 	}
 
 	/**
-	 * Receives the path of the configuration.properties file in its constructor.
-	 * 
-	 * @param config
+	 * Receives the path of the .properties file in its constructor from a resource.
 	 */
-	public MongoDBAASAPI(BaSyxMongoDBConfiguration config, String aasId, MongoClient client) {
-		this.setConfiguration(config, client);
-		this.setAASId(aasId);
-		configureIndexForAasId(mongoOps);
+	public MongoDBAASAPI(String resourceConfigPath, String shellIdentificationId, MongoClient client) {
+		this(MongoDBBaSyxStorageAPIFactory.<AssetAdministrationShell>create(configFromResource(resourceConfigPath).getSubmodelCollection(), AssetAdministrationShell.class, configFromResource(resourceConfigPath), client), shellIdentificationId);
 	}
 
-	private void configureIndexForAasId(MongoOperations mongoOps) {
-		TextIndexDefinition idIndex = TextIndexDefinition.builder().onField(AASIDPATH).build();
-		mongoOps.indexOps(AssetAdministrationShell.class).ensureIndex(idIndex);
+	/**
+	 * Constructor using default MongoDB connections
+	 */
+	public MongoDBAASAPI(String shellIdentificationId, MongoClient client) {
+		this(DEFAULT_CONFIG_PATH, shellIdentificationId, client);
 	}
 
 	/**
-	 * Receives the path of the .properties file in its constructor from a resource.
+	 * Receives the path of the configuration.properties file in its constructor.
 	 * 
+	 * @param config
 	 * @deprecated Use the new constructor using a MongoClient
 	 */
 	@Deprecated
-	public MongoDBAASAPI(String resourceConfigPath, String aasId) {
-		config = new BaSyxMongoDBConfiguration();
-		config.loadFromResource(resourceConfigPath);
-		this.setConfiguration(config);
-		this.setAASId(aasId);
+	public MongoDBAASAPI(BaSyxMongoDBConfiguration config, String shellIdentificationId) {
+		this(MongoDBBaSyxStorageAPIFactory.<AssetAdministrationShell>create(config.getAASCollection(), AssetAdministrationShell.class, config), shellIdentificationId);
```

Comment:
MongoDBBaSyxStorageAPIFactory.<AssetAdministrationShell>create should be MongoDBBaSyxStorageAPIFactor<AssetAdministrationShell>.create (note different position of ".")

Reply:
This is also at other places